src/Header/DownloadStream.h

```diff
@@ -25,7 +26,9 @@ class DownloadStream {
   // filename include file path
   std::string filename;
   // default chunk size
-  unsigned long chuck_size;
+  unsigned long chunk_size;
+  // True indicate that it has the read access of the file, otherwise.
+  bool reading;
```

Comment:
[Optional] std::optional<std::fstream> can be used here instead for one lesser variable.

Reply:
Removed. Using fd.fail() to check badbit and failbit.